providers/cstor/cstor-disk-pool/test.yml

```diff
@@ -41,38 +41,98 @@
-          shell: >
-            kubectl get pod {{ item }} -n {{ operator_ns }}
-            -o=jsonpath='{range .status.containerStatuses[*]}{.state}{"\n"}{end}' |
-            grep -w running | wc -l
-          args:
-            executable: /bin/bash
-          register: runningStatusCount
-          with_items: "{{ cstor_pool_pod.stdout_lines }}"
-          until: "runningStatusCount.stdout == \"2\""
-          delay: 30
-          retries: 10
+            - name: Verify if cStor Disk Pool are created on each node
+              shell: > 
+                 kubectl get pods -n {{ operator_ns }} -l openebs.io/storage-pool-claim={{ pool_name }}
+                 --no-headers -o custom-columns=:status.phase
+              args:
+                executable: /bin/bash
+              register: pool_count
+              until: pool_count.stdout_lines|length == maxpool_count.stdout|int
+              retries: 10
+              delay: 20
+
+            - name: Get cStor Disk Pool names to verify the container statuses
+              shell: >
+                kubectl get pods -n {{ operator_ns }} -l openebs.io/storage-pool-claim={{ pool_name }} 
+                --no-headers -o=custom-columns=NAME:".metadata.name"
+              args:
+                executable: /bin/bash
+              register: cstor_pool_pod
+
+            - name: Get the runningStatus of pool pod
+              shell: >
+                kubectl get pod {{ item }} -n {{ operator_ns }}
+                -o=jsonpath='{range .status.containerStatuses[*]}{.state}{"\n"}{end}' |
+                grep -w running | wc -l
+              args:
+                executable: /bin/bash
+              register: runningStatusCount
+              with_items: "{{ cstor_pool_pod.stdout_lines }}"
+              until: "runningStatusCount.stdout == \"2\""
+              delay: 30
+              retries: 10
+
+          when: lookup('env', 'ACTION') == 'create'    
+
+        - block:
+                
+            - name: Delete cStor Disk Pool
+              shell: kubectl delete -f "{{ playbook_dir }}/{{ cstor_spc }}"
+              args:
+                executable: /bin/bash
+
+            - name: Verify if cStor Disk Pools are deleted on each node
+              shell: >
+                 kubectl get pods -n {{ operator_ns }} -l openebs.io/storage-pool-claim={{ pool_name }}
+              args:
+                executable: /bin/bash
+              register: pool_verify
+              until: "'No resources found' in pool_verify.stderr"
+              retries: 10
+              delay: 15
+
+            - name: Verify the sp and csp are deleted
+              shell: >
+                kubectl get {{ item }} -l openebs.io/storage-pool-claim={{pool_name}} 
+                --no-headers -o=custom-columns=NAME:".metadata.name" 
+              args:
+                executable: /bin/bash
+              register: sp_and_csp
+              until: "sp_and_csp.rc == 0" 
```

Comment:
This condition also has to be modified. The return code will be 0 even if the resources are not deleted.